### PR TITLE
xplat cli role list command changes for custom roles

### PR DESCRIPTION
**Pull request by shuagarw**
xplat cli role list command changes for custom roles

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@shuagarw__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (shuagarw). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/role/role.definition._js [hunk 1]
```diff
@@ -25,35 +25,44 @@ var $ = utils.getLocaleString;
 
 exports.init = function (cli) {
   var log = cli.output;
-
   var role = cli.category('role')
     .description($('Commands to manage your Azure roles'));
+  var CUSTOM_ROLE_TYPE = 'CustomRole';
```
Comment (amarzavery) [inline lib/commands/arm/role/role.definition._js]: can you move this variable declaration up after var log = cli.output;

File: lib/commands/arm/role/role.definition._js [hunk 2]
```diff
@@ -25,35 +25,44 @@ var $ = utils.getLocaleString;
...
 
   role.command('list')
+    .option('--custom', $('If given, display only the custom roles instead of all role definitions'))
+    .option('--subscription <subscription>', $('Subscription id or name of where the role assignment is from.'))
```
Comment (amarzavery) [inline lib/commands/arm/role/role.definition._js]: what does this mean 'or name of where the role assignment is from.' ?

**Comment by amarzavery** [comment]
@shuagarw - How will the end user create custom roles?

File: lib/commands/arm/role/role.definition._js (hunk 2 above)
Comment (shuagarw) [inline lib/commands/arm/role/role.definition._js]: Actually it should be 'role definition is from'. I will commit this change in same branch.
Basically it tells that look up for this role in some other subscription scope. Like I have created a role assignment in subA, but when I am in a different subscription (lets say 'subB' of same tenant) on CLI,  and running the coming to fetch this role will not fetch it unless I specify --subscription <SubA>. I picked the description of this param from existing RoleAssignment commands. We can modify them, once we complete the commands, after discussing it with the PMs.

**Comment by shuagarw** [comment]
That is currently done by powershell. Namrata will send the create command pull request soon.

**Pull request merged**